addons/survey/models/survey_survey.py

```diff
@@ -517,6 +517,23 @@ def _get_next_page_or_question(self, user_input, page_or_question_id, go_back=Fa
 
         return Question
 
+    def _is_last_page_or_question(self, user_input, page_or_question):
+        pages_or_questions = self._get_pages_or_questions(user_input)
+        current_page_index = pages_or_questions.ids.index(page_or_question.id)
+        next_page_or_question_candidates = pages_or_questions[current_page_index + 1:]
```

Comment:
Would create a variable `is_last_question = True` before searching and return that.